tests/cli/test_register.py

```diff
@@ -1,9 +1,65 @@
 from flask import current_app
 import pytest
 from core.constants import REGISTERED_CANDIDATE
-from computable.helpers.transaction import call
+from computable.helpers.transaction import call, transact
+
+def test_register(w3, ether_token, market_token, voting, parameterizer, reserve, datatrust, ctx):
+    user = w3.eth.defaultAccount 
+    user_bal = call(ether_token.balance_of(user))
+    assert user_bal == 0
+
+    # Deposit ETH in EtherToken
+    tx = transact(ether_token.deposit(
+        w3.toWei(10, 'ether'), {'from': user}))
+    rct = w3.eth.waitForTransactionReceipt(tx)
+    new_user_bal = call(ether_token.balance_of(user))
+    assert new_user_bal == w3.toWei(10, 'ether')
+    assert rct['status'] == 1
+
+    # Approve the spend
+    old_allowance = call(ether_token.allowance(user, reserve.address))
+    assert old_allowance == 0
+    tx= transact(ether_token.approve(reserve.address, w3.toWei(10, 'ether'), opts={'from': user}))
+    rct = w3.eth.waitForTransactionReceipt(tx)
+    assert rct['status'] == 1
+    new_allowance = call(ether_token.allowance(user, reserve.address))
+    assert new_allowance == w3.toWei(10, 'ether')
+
+    # Perform pre-checks for support 
+    support_price = call(reserve.get_support_price())
+    assert new_user_bal >= support_price
+    assert new_allowance >= new_user_bal
+    minted = (new_user_bal // support_price) * 10**9
+    assert minted == 10**7 * w3.toWei(1, 'gwei')
+    priv = call(market_token.has_privilege(reserve.address))
+    assert priv == True
+    total_supply = call(market_token.total_supply())
+    assert total_supply == w3.toWei(4, 'ether')
+
+    # Call support
+    tx = transact(reserve.support(new_user_bal, opts={'gas': 1000000, 'from': user}))
+    rct = w3.eth.waitForTransactionReceipt(tx)
+    assert rct['status'] == 1
+    logs = reserve.deployed.events.Supported().processReceipt(rct)
+    cmt_user_bal = call(market_token.balance_of(user))
+    # There is the creator already
+    assert cmt_user_bal >= w3.toWei(10, 'milliether')
+    new_supply = call(market_token.total_supply())
+    assert new_supply == total_supply + w3.toWei(10, 'milliether')
+
+    stake = call(parameterizer.get_stake())
+    assert stake <= cmt_user_bal
+
+    # Approve the market token allowance
+    old_mkt_allowance = call(market_token.allowance(user, voting.address))
+    assert old_mkt_allowance == 0
+    tx = transact(market_token.approve(voting.address, w3.toWei(10, 'milliether'), opts={'from': user}))
+    rct = w3.eth.waitForTransactionReceipt(tx)
+    assert rct['status'] == 1
+    new_mkt_allowance = call(market_token.allowance(user, voting.address))
+    assert new_mkt_allowance == w3.toWei(10, 'milliether')
+    assert stake <= new_mkt_allowance
```

Comment:
this is what i was talking about.

move all steps for setting up the register test to smaller tests that run before it. one very large test will be hard to refactor once we setup a helper or change the conftest.

you can be as piecemeal as you want with separate tests for `test_has_foo`, `test_can_bar` or a larger one that encompasses a logical outcome like `test_can_stake` but don't leave it all in the `register` test

Reply:
Factored out into smaller tests